configure.ac

```
@@ -37,7 +37,14 @@ AM_CONDITIONAL([INSTALL_MACOS_FS_LINK], [case $build_os in darwin* ) test x"$ena
 # _DARWIN_BETTER_REALPATH fixes MacOS realpath() broken around Catalina (#83).
 my_CPPFLAGS="-D_REENTRANT -D_FILE_OFFSET_BITS=64 -D_XOPEN_SOURCE=700 -D__BSD_VISIBLE=1 -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_DARWIN_BETTER_REALPATH"
 
-my_CFLAGS="-std=c99 -Wall -Wextra -Wpedantic -fno-common"
+dnl libfuse >=3.17 started requiring gnu11 from the C compiler
+AS_IF([test "x$with_fuse2" == "xyes"], [
```

`==` is not POSIX shell, and `configure` scripts run w/ `/bin/sh` shebang, so we need:
```suggestion
AS_IF([test "x$with_fuse2" = "xyes"], [
```

---

@thesamesam good point! I copy-pasted from further down, this needs fixing in multiple (existing) places:

```console
# git --no-pager grep == configure.ac
configure.ac:if test x"$enable_debug_output" == "xyes" ; then
configure.ac:if test x"$with_core_foundation" == "xyes" ; then
configure.ac:AS_IF([test "x$with_fuse2" == "xyes"], [
configure.ac:AS_IF([test "x$with_fuse2" == "xyes"],
configure.ac:    [AS_IF([test "x$with_fuse3" == "xyes"],
configure.ac:        [AS_IF([test "x$with_fuse_t" == "xyes"],
```

And then CI could run configure with Dash to expose issues like that before introduction. Actually that's something I should consider doing in multiple other projects also.

---

Ubuntu and Debian build their `dash` with `LINENO` disabled so that GNU Autoconf's `configure` detects that `LINENO` is missing and it'll fallback to bash.

---

> Ubuntu and Debian build their `dash` with `LINENO` disabled so that GNU Autoconf's `configure` detects that `LINENO` is missing and it'll fallback to bash.

@thesamesam tu be sure I'm reading this right: are you saying that autoconf will put a Bash shebang line into `configure` in that case?

> However, this said, configure "errors" that manifest with non-bash are usually non-fatal anyway, just result in warnings and misdetection of features.

Understood, so it will not be waterproof and just detect some issues but not all.

---

> Ubuntu and Debian build their `dash` with `LINENO` disabled so that GNU Autoconf's `configure` detects that `LINENO` is missing and it'll fallback to bash.

@thesamesam update: I found…

```console
$ head -n1 configure
#! /bin/sh
```

… with image `ubuntu-24.04` in GitHub Actions. From what you said, I was expecting to see Bash. Did I misunderstand?

---

The shebang is always `#!/bin/sh`. If `configure` detects that the running shell lacks `LINENO`, it will re-exec with bash.

---

@thesamesam interesting. I'm reading that as "running through Dash would help on systems building Dash with LINENO support but not elsewhere, which includes Debian and Ubuntu."